lib/mail_xoauth2/imap_xoauth2_authenticator.rb

```diff
@@ -21,4 +21,8 @@ def initialize(user, oauth2_token)
   end
 end
 
-Net::IMAP::SASL.add_authenticator('XOAUTH2', MailXoauth2::ImapXoauth2Authenticator)
+if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('3.3.3')
```

Comment:
IMO this shouldn't inquire `Gem::Version.new(RUBY_VERSION)` but rather `Gem::Version.new(Net::IMAP::VERSION)`.

"net-imap" is [bundled gem](https://stdgems.org/net-imap/), yes. Still, it's absolutely permitted to use a different (newer) version comparing to one that's available with Ruby installation.

Any "net-imap" version >= 0.4.x would contain [this](https://github.com/ruby/net-imap/commit/0e1e416e04f88baa3fdf4a6b2e171f749d41f4e6) commit and hence usable with `Net::IMAP::SASL.add_authenticator` call.

```suggestion
if Gem::Version.new(Net::IMAP::VERSION) >= Gem::Version.new('0.4.0')
```